azure/test.yml

```diff
@@ -34,23 +34,23 @@ jobs:
     parameters:
       rust: ${{ parameters.rust }}
       setup: ${{ parameters.setup }}
-  - ${{ if and(eq(parameters.single_threaded, 'true'), eq(parameters.test_ignored, 'true')) }}:
-    - script: cargo test --all --features "${{ parameters.features }}" -- --ignored --test-threads=1
-      displayName: Run tests (single-threaded w/ignored)
+  - ${{ if ne(parameters.single_threaded, 'true') }}:
+    - script: cargo test --features "${{ parameters.features }}" --all
+      displayName: Run tests
       env:
         ${{ insert }}: ${{ parameters.envs }}
-  - ${{ if and(eq(parameters.single_threaded, 'true'), ne(parameters.test_ignored, 'true')) }}:
+  - ${{ if eq(parameters.single_threaded, 'true') }}:
     - script: cargo test --all --features "${{ parameters.features }}" -- --test-threads=1
       displayName: Run tests (single-threaded)
       env:
         ${{ insert }}: ${{ parameters.envs }}
   - ${{ if and(ne(parameters.single_threaded, 'true'), eq(parameters.test_ignored, 'true')) }}:
     - script: cargo test --all --features "${{ parameters.features }}" -- --ignored
-      displayName: Run tests (incl. ignored)
+      displayName: Run ignored tests
       env:
         ${{ insert }}: ${{ parameters.envs }}
-  - ${{ if and(ne(parameters.single_threaded, 'true'), ne(parameters.test_ignored, 'true')) }}:
-    - script: cargo test --features "${{ parameters.features }}" --all
-      displayName: Run tests
+  - ${{ if and(eq(parameters.single_threaded, 'true'), eq(parameters.test_ignored, 'true')) }}:
```

Comment:
I think it'd be _slightly_ easier to read this and the one above if the `test_ignored` check came first:
```
${{ if and(eq(parameters.test_ignored, 'true'), eq(parameters.single_threaded, 'true')) }}:
```
Otherwise LGTM :+1:

Reply:
Good point, fixed!